_posts/2024-10-03-pixi-basics.md

```diff
@@ -0,0 +1,358 @@
+
+Benchmark 1: pixi add samtools && pixi install && pixi remove samtools
+
+  Time (mean ± σ):     24.693 s ±  0.831 s    [User: 7.415 s, System: 8.035 s]
+  Range (min … max):   24.182 s … 25.921 s    4 runs
+ 
+Benchmark 2: conda create -y -p condaenv samtools && rm -rf condaenv
+
+  Time (mean ± σ):     35.740 s ±  0.770 s    [User: 6.519 s, System: 5.505 s]
+  Range (min … max):   34.609 s … 36.290 s    4 runs
+ 
+Summary
+  'pixi add samtools && pixi install && pixi remove samtools' ran
+    1.45 ± 0.06 times faster than 'conda create -y -p condaenv samtools && rm -rf condaenv'
+
+Benchmark 1: pixi add vadr && pixi install && pixi remove vadr
+
+  Time (mean ± σ):     70.947 s ±  0.554 s    [User: 23.427 s, System: 37.370 s]
+  Range (min … max):   70.519 s … 71.733 s    4 runs
+ 
+Benchmark 2: conda create -y -p condaenv vadr && rm -rf condaenv
+
+  Time (mean ± σ):     127.758 s ±  7.067 s    [User: 18.144 s, System: 16.139 s]
+  Range (min … max):   122.523 s … 137.886 s    4 runs
+ 
+Summary
+  'pixi add vadr && pixi install && pixi remove vadr' ran
+    1.80 ± 0.10 times faster than 'conda create -y -p condaenv vadr && rm -rf condaenv'
+
+Benchmark 1: pixi add ska && pixi install && pixi remove ska
+
+  Time (mean ± σ):      5.076 s ±  0.149 s    [User: 5.854 s, System: 1.689 s]
+  Range (min … max):    4.948 s …  5.283 s    4 runs
+ 
+Benchmark 2: conda create -y -p condaenv ska && rm -rf condaenv
+
+  Time (mean ± σ):      8.831 s ±  0.466 s    [User: 4.585 s, System: 2.413 s]
+  Range (min … max):    8.215 s …  9.262 s    4 runs
+ 
+Summary
+  'pixi add ska && pixi install && pixi remove ska' ran
+    1.74 ± 0.10 times faster than 'conda create -y -p condaenv ska && rm -rf condaenv'
+
+Benchmark 1: pixi add pangolin && pixi install && pixi remove pangolin
+
+  Time (mean ± σ):     160.734 s ±  5.329 s    [User: 41.234 s, System: 105.102 s]
+  Range (min … max):   154.892 s … 167.771 s    4 runs
+ 
+Benchmark 2: conda create -y -p condaenv pangolin && rm -rf condaenv
+
+  Time (mean ± σ):     321.362 s ±  5.470 s    [User: 31.665 s, System: 39.065 s]
+  Range (min … max):   315.114 s … 328.384 s    4 runs
+ 
+Summary
+  'pixi add pangolin && pixi install && pixi remove pangolin' ran
+    2.00 ± 0.07 times faster than 'conda create -y -p condaenv pangolin && rm -rf condaenv'
+Benchmark 1: pixi add ivar && pixi install && pixi remove ivar
+
+  Time (mean ± σ):     24.782 s ±  0.354 s    [User: 7.629 s, System: 7.915 s]
+  Range (min … max):   24.505 s … 25.275 s    4 runs
+ 
+Benchmark 2: conda create -y -p condaenv ivar && rm -rf condaenv
+
+  Time (mean ± σ):     37.005 s ±  0.452 s    [User: 6.882 s, System: 5.673 s]
+  Range (min … max):   36.575 s … 37.606 s    4 runs
+ 
+Summary
+  'pixi add ivar && pixi install && pixi remove ivar' ran
+    1.49 ± 0.03 times faster than 'conda create -y -p condaenv ivar && rm -rf condaenv'
+
+```
+
+</details>
+
+For the complicated installation, pixi is 2.21x faster than conda.
```

Comment:
Is this compared to a conda version that runs mamba under the hood (probably, right?)? Perhaps worth pointing out. Would expect even greater speedups over the old conda, which some people might have in mind.

Reply:
I added the line 

> For Conda, I used [miniforge](https://github.com/conda-forge/miniforge) v24.7.1-2.